### PR TITLE
feat: render secret value when using dynamic definition

### PR DESCRIPTION
**Pull request by donch1989**
Because

- In the dynamic component definition, we might need to use the secret value to generate the dynamic definition. Therefore, we need to render the secret value for this.

This commit

- Renders the secret value when using dynamic definition.

### CODE REVIEW

**Comment by donch1989** [comment]
Hi @chuang8511,
Please check this PR.

File: pkg/service/pipeline.go [hunk 1]
```diff
@@ -631,7 +631,7 @@ func (s *service) preTriggerPipeline(ctx context.Context, isAdmin bool, ns resou
 	for {
 		var nsSecrets []*datamodel.Secret
 		// TODO: should use ctx user uid
-		nsSecrets, _, pt, err = s.repository.ListNamespaceSecrets(ctx, ns.Permalink(), 100, "", filtering.Filter{})
+		nsSecrets, _, pt, err = s.repository.ListNamespaceSecrets(ctx, ns.Permalink(), 100, pt, filtering.Filter{})
 		if err != nil {
```
Comment (chuang8511) [inline pkg/service/pipeline.go:635]: NIT: I am not quite sure why we need this part in this PR.
I will trace code more to catch up codebase later.

File: pkg/service/convert.go [hunk 2]
```diff
@@ -141,14 +145,43 @@ func (c *converter) convertResourcePermalinkToName(ctx context.Context, rsc any)
 	return nil
 }
 
-func (c *converter) includeComponentDetail(ctx context.Context, compConfig *componentbase.ComponentConfig, useDynamicDef bool) error {
+func (c *converter) processSetup(ctx context.Context, ownerPermalink string, setup map[string]any) map[string]any {
+	rendered := map[string]any{}
+	for k, v := range setup {
+		switch v := v.(type) {
+		case map[string]any:
+			rendered[k] = c.processSetup(ctx, ownerPermalink, v)
```
Comment (chuang8511) [inline pkg/service/convert.go:153]: Question:
I checked archetypeap, huggingface, openai.
They seem to be only 1 layer when we setup those components.
I am not sure if it is common thing to have several map in a setup.
If not common, maybe it is too overkilling?

**Review by chuang8511 — COMMENTED** [review]
Mainly LGTM. Only have a question.
Thanks for helping me do it!
I will directly use this branch to move on my development.

File: pkg/service/convert.go [hunk 3]
```diff
@@ -141,14 +145,43 @@ func (c *converter) convertResourcePermalinkToName(ctx context.Context, rsc any)
...
+		case string:
+			if strings.HasPrefix(v, "${"+recipe.SegSecret+".") && strings.HasSuffix(v, "}") {
```
Comment (chuang8511) [inline pkg/service/convert.go:155]: Important: I think I will need a else case here.

**Review by chuang8511 — COMMENTED** [review]
I will add it first for my testing!

File: pkg/service/pipeline.go (hunk 1 above)
Comment (donch1989) [reply pkg/service/pipeline.go:635]: It is just a small bug-fix, I put this in the same PR.

File: pkg/service/convert.go (hunk 3 above)
Comment (donch1989) [reply pkg/service/convert.go:155]: Yeah, you are correct. Fixed.

File: pkg/service/convert.go (hunk 2 above)
Comment (donch1989) [reply pkg/service/convert.go:153]: Yeah, we don't have this case for now, but adding it doesn't hurt.

**Pull request merged**